### ISSUE TITLE
Saving indicator is not showing, even though changes are saved.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1667]
In Chrome, when editing a project, changes are saved even though the Saving indicator never shows up...

![saving](https://cloud.githubusercontent.com/assets/25212/22270926/e519eeec-e246-11e6-8966-a21cd5f93159.gif)

This change survived refreshing the browser, and also closing the tab and re-opening the project.

**Comment by humphd** [comment]
The issue here is that we have two kinds of saving:
* saving the files to "disk" in IndexedDB in the browser so they will be available locally and survive a refresh
* sending the files to the server to be saved there.  You have to be logged in for this to happen.

I'm not sure how much we want to expose the user to this difference.

**Comment by flukeout** [comment]
I'm not seeing a "Saving.." indicator in either of the cases above, one of which I'm guessing sends the files to the server as I'm closing the browser tab and opening the project from the project list.

When do we currently attempt to show the "Saving..." indicator? I was under the impression it was only for server saves.

**Comment by Taguiar** [comment]
I wold like to fix this bug

**Comment by flukeout** [comment]
Sounds good @Taguiar

**Comment by gideonthomas** [comment]
@flukeout yes that indicator should only show up if it saves to the server.

I am hesitant to leave this as a "good first bug" since this might involve touching/investigating complex code in our file syncing logic. @humphd what do you think?

**Comment by Pomax** [comment]
As long as the code path is easy to trace and we're available for questions, I don't think this is too complicated a bug to work on. I'd be in favour of the mantra "see if it's too much work, switch if it is": if more than a day is required to get *into* the bug, then this is a bad bug, but ruling it out before someone starts to look into it is probably acting too early... someone might be really good at bug hunting and love this challenge!

**Comment by humphd** [comment]
The code for this (at least the code that shows/hides things) is here:

https://github.com/mozilla/thimble.mozilla.org/blob/fa3dde088dc016a562520e62a9df1f5558ba8f70/public/editor/scripts/editor/js/fc/filesystem-sync.js#L37

The code that's triggering those events is here:

https://github.com/mozilla/thimble.mozilla.org/blob/fa3dde088dc016a562520e62a9df1f5558ba8f70/public/editor/scripts/editor/js/fc/sync-manager.js#L468-L484

**Comment by Taguiar** [comment]
I found the bug really fast, but the problem I am having is thaT I can log in from my local computer.
What you think @humphd

**Comment by Taguiar** [comment]
@flukeout Are you logged in your account? Because thimble only saves your work if you are logged in.

**Comment by Taguiar** [comment]
I got this bug fixed. This was was really funny to do since the code was working perfectly in some operating systems/browsers and some nothing was appearing at all. My changes were focused when the user is logged in. But I think that it would be interesting if some kind of prompt would pop up even when the user is not logged in (the save is done locally). The pull request will be done this afternoon!

**Comment by humphd** [comment]
@Taguiar can you file a new bug on the "Saving..." and "Saved." stuff for the case when users aren't logged in?  I think that's a neat idea, and we should follow-up on that, but not in this bug.
